### PR TITLE
Added API documentation, babel-react-gettext-parser example

### PR DESCRIPTION
**Pull request by iszak**

### CODE REVIEW

**Comment by iszak** [comment]
Rebased this and updated it as per the renamed methods

**Comment by alexanderwallin** [comment]
This is a good initiative. I wonder whether we should set up some doc generation while where at it? The only one I've successfully used previously is [jsdoc](http://usejsdoc.org/), although I'm not very happy with it.

Thoughts?

**Comment by iszak** [comment]
@alexanderwallin I like the idea because it's all too easy to have documentation out of sync with the actual implementation and so having them close together seems like a good idea.

I would need to have a look around at solutions (what does the react community use?) as I don't know of any de facto standard (besides jsdoc which is more general javascript). I don't think we should hold back this ticket as it may take some time to investigate and find a solution and in that time it would be better to have some documentation.

Shall we merge this and create a ticket which explores which features e.g. website generation we want and then possible candidates?

**Comment by iszak** [comment]
I would like to document what `tcnp` means as I still find it difficult to remember what each letter maps to what functionality. I assume they relate to the first letter of the meaning, are you able to clarify them?

t = translate (message parameter)
c = interpolate (scope parameter)
n = plural (one, other, count, parameter)
p = context

**Comment by alexanderwallin** [comment]
> Shall we merge this and create a ticket which explores which features e.g. website generation we want and then possible candidates?

Good point, let's go with that.

> t = translate (message parameter)
> c = interpolate (scope parameter)
> n = plural (one, other, count, parameter)
> p = context

That's all correct. `c` was intended to refer to "component". In hindsight, perhaps `i` would've been more clear.

File: README.md [hunk 1]
```diff
@@ -128,10 +128,76 @@ function PotatoNotification({ notificationCode, t }) {
...
+
+Translate a plural message
+
+### tp(context, message)
+
+Translate a message and context
+
+### tnp(context, one, other, count)
+
+Translate a plural message with context and variable interpolate
```
Comment (alexanderwallin) [inline README.md]: I think it would definitely help to clarify this in method descriptions, since it is a gettext library.

Comment (alexanderwallin) [inline README.md]: There's no interpolation in `tnp()`, though!

File: README.md [hunk 2]
```diff
@@ -128,10 +128,76 @@ function PotatoNotification({ notificationCode, t }) {
 export default withTranslators(PotatoNotification)
 ```
 
+## Via [`babel-plugin-react-gettext-parser`](http://github.com/alexanderwallin/babel-plugin-react-gettext-parser)
```
Comment (alexanderwallin) [inline README.md]: Nice to have a preset config! Make it an `###` so that it falls under the *Usage*. 👍

Comment (iszak) [inline README.md]: Done

**Comment by alexanderwallin** [comment]
Nice one!

**Pull request merged**